### ISSUE TITLE
EZMicrophone set custom AudioStreamBasicDescription

### ISSUE THREAD

**Issue by icanswiftabit** [syedhali/EZAudio#7]
Great class. I struggle with AudioUnit for some time, but this class is great. :) But for me, one thing is probably missing or I just looking for it in wrong place. Like in subject, it's possible to set custom AudioStreamBasicDescription for EZMicrophone class? Because I can't find metho/way for doing it. If there is not, what is the reason? Not enought time or someother.

**Comment by vfxdrummer** [comment]
Hey Syed -

First of all, amazing framework! This is a bit of a generic comment, but it seems that whenever I try to override the ASBD on the microphone I get errors like :

Error: Failed to fill complex buffer in float converter ('insz')

  Here are example settings :
  audioStreamBasicDescription.mFormatID          = kAudioFormatLinearPCM;
  audioStreamBasicDescription.mSampleRate        = 16000;
  audioStreamBasicDescription.mFramesPerPacket   = 1;
  audioStreamBasicDescription.mBytesPerPacket    = 2; //16 bits * 1 channel
  audioStreamBasicDescription.mBytesPerFrame     = 2;
  audioStreamBasicDescription.mChannelsPerFrame  = 1; //1 channel
  audioStreamBasicDescription.mBitsPerChannel    = 16;
  audioStreamBasicDescription.mFormatFlags       = kAudioFormatFlagIsSignedInteger |
  kAudioFormatFlagIsBigEndian | kAudioFormatFlagIsPacked;
  audioStreamBasicDescription.mReserved          = 0;

Basically, I am trying to make the stream 16 bits / 16 kHz so that I can upload it to  Nuance's speech to text dictation.

Any thoughts on this? Thanks again!
